Updates to env var names (#3134)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/about-state.md

State and defer can be set by environment variables as well as CLI flags:

<VersionBlock lastVersion="1.4">

- `--state` or `DBT_ARTIFACT_STATE_PATH`: file path
- `--defer` or `DBT_DEFER_TO_STATE`: boolean

</VersionBlock>

<VersionBlock firstVersion="1.5">

- `--state` or `DBT_STATE`: file path
- `--defer` or `DBT_DEFER`: boolean

:::warning Syntax deprecated

> The original syntax for state (`DBT_ARTIFACT_STATE_PATH`) and defer (`DBT_DEFER_TO_STATE`) have been deprecated in dbt v1.5. Backward compatibility is supported in this version but will be removed in an as-of-yet-undetermined future release.

Reviewer: this is not a blocker and merely a suggestion on switching to active voice. it may be an instance where passive is ok per style guide.

```suggestion
In dbt v1.5, we deprecated the original syntax for state (`DBT_ARTIFACT_STATE_PATH`) and defer (`DBT_DEFER_TO_STATE`). Although dbt supports backward compatibility with the old syntax, we will remove it in a future release that we have not yet determined.
```
